File: pegasusX/apps/driver-app-ios/driverappios/driverappios/Views/Components/Home/RemainingStopsStepper.swift
import SwiftUI

struct RemainingStopsStepper: View {
    let stops: [RemainingStop]
    var onSelect: ((String) -> Void)?

    var body: some View {
        VStack(alignment: .leading, spacing: 10) {
            Text("REMAINING STOPS")
                .font(.system(size: 11, weight: .heavy, design: .monospaced))
                .foregroundStyle(LabTheme.fgTertiary)
            if stops.isEmpty {
                Text("No remaining stops")
                    .font(.system(size: 13, weight: .medium))
                    .foregroundStyle(LabTheme.fgTertiary)
            } else {
                ForEach(Array(stops.enumerated()), id: \.element.id) { index, stop in
                    Button {
                        onSelect?(stop.id)
                    } label: {
                        HStack(alignment: .top, spacing: 10) {
                            VStack(spacing: 0) {
                                Circle()
                                    .fill(dotColor(stop))
                                    .frame(width: 10, height: 10)
                                if index < stops.count - 1 {
                                    Rectangle()
                                        .fill(LabTheme.separator.opacity(0.35))
                                        .frame(width: 2, height: 22)
                                }
                            }
                            VStack(alignment: .leading, spacing: 2) {
                                Text(stop.title.isEmpty ? stop.id : stop.title)
                                    .font(.system(size: 14, weight: .semibold))
                                    .foregroundStyle(LabTheme.fg)
                                    .multilineTextAlignment(.leading)
                                Text(stop.state.label)
                                    .font(.system(size: 11, weight: .bold, design: .monospaced))
                                    .foregroundStyle(stop.firstClass ? LabTheme.destructive : LabTheme.fgSecondary)
                            }
                            Spacer(minLength: 0)
                        }
                    }
                    .buttonStyle(.plain)
                    .accessibilityLabel("\(stop.title), \(stop.state.label)")
                }
            }
        }
        .padding(LabTheme.s16)
        .labCard()
    }

    private func dotColor(_ stop: RemainingStop) -> Color {
        if stop.state == .FISCAL_FAILED { return LabTheme.destructive }
        if stop.state == .ARRIVED_SHOP_CLOSED { return LabTheme.warning }
        return LabTheme.fgSecondary
    }
}

struct FieldMoneyStrip: View {
    let counts: MoneyHealthCounts

    var body: some View {
        HStack(spacing: 8) {
            strip("CASH", counts.pendingCash)
            strip("FISCAL", counts.openFiscal)
            strip("CREDIT", counts.creditLeave)
        }
    }

    private func strip(_ label: String, _ count: Int) -> some View {
        VStack(alignment: .leading, spacing: 2) {
            Text(label)
                .font(.system(size: 10, weight: .heavy, design: .monospaced))
                .foregroundStyle(LabTheme.fgTertiary)
            Text(count == 0 ? "empty" : "\(count)")
                .font(.system(size: 13, weight: .semibold))
                .foregroundStyle(count == 0 ? LabTheme.fgTertiary : LabTheme.fg)
        }
        .frame(maxWidth: .infinity, alignment: .leading)
        .padding(10)
        .background(LabTheme.card, in: RoundedRectangle(cornerRadius: LabTheme.buttonRadius, style: .continuous))
    }
}

struct ManifestBulletMeter: View {
    let state: String?
    let usedVU: Double?
    let maxVU: Double

    var body: some View {
        VStack(alignment: .leading, spacing: 6) {
            Text("MANIFEST")
                .font(.system(size: 11, weight: .heavy, design: .monospaced))
                .foregroundStyle(LabTheme.fgTertiary)
            Text((state?.isEmpty == false ? state! : "unavailable"))
                .font(.system(size: 13, weight: .bold, design: .monospaced))
                .foregroundStyle(LabTheme.fg)
            if let used = usedVU, maxVU > 0 {
                ProgressView(value: min(max(used / maxVU, 0), 1))
                Text(String(format: "%.0f / %.0f VU", used, maxVU))
                    .font(.system(size: 11, weight: .medium, design: .monospaced))
                    .foregroundStyle(LabTheme.fgSecondary)
            } else {
                Text("VU unavailable")
                    .font(.system(size: 11, weight: .medium))
                    .foregroundStyle(LabTheme.fgTertiary)
            }
        }
        .padding(LabTheme.s16)
        .labCard()
    }
}

File: pegasusX/apps/driver-app-android/app/src/main/java/com/pegasusx/driver/ui/screens/home/components/RemainingStopsStepper.kt
package com.pegasusx.driver.ui.screens.home.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.LinearProgressIndicator
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.pegasusx.driver.data.model.OrderState
import com.pegasusx.driver.ui.components.PegasusCard
import com.pegasusx.driver.ui.screens.home.MoneyHealthCounts
import com.pegasusx.driver.ui.screens.home.RemainingStop
import com.pegasusx.driver.ui.theme.LocalPegasusColors
import com.pegasusx.driver.ui.theme.PegasusSpacing
import com.pegasusx.driver.ui.theme.pressable

@Composable
fun RemainingStopsStepper(
    stops: List<RemainingStop>,
    onSelect: (String) -> Unit = {},
    modifier: Modifier = Modifier,
) {
    val lab = LocalPegasusColors.current
    PegasusCard(modifier = modifier.fillMaxWidth()) {
        Column(
            modifier = Modifier.padding(PegasusSpacing.s16),
            verticalArrangement = Arrangement.spacedBy(10.dp),
        ) {
            Text(
                text = "REMAINING STOPS",
                fontSize = 11.sp,
                fontWeight = FontWeight.Black,
                fontFamily = FontFamily.Monospace,
                color = lab.fgTertiary,
            )
            if (stops.isEmpty()) {
                Text("No remaining stops", fontSize = 13.sp, color = lab.fgTertiary)
            } else {
                stops.forEachIndexed { index, stop ->
                    Row(
                        modifier = Modifier
                            .fillMaxWidth()
                            .pressable { onSelect(stop.id) },
                        verticalAlignment = Alignment.Top,
                    ) {
                        Column(horizontalAlignment = Alignment.CenterHorizontally) {
                            Box(
                                modifier = Modifier
                                    .size(10.dp)
                                    .clip(CircleShape)
                                    .background(
                                        when (stop.state) {
                                            OrderState.FISCAL_FAILED -> lab.destructive
                                            OrderState.ARRIVED_SHOP_CLOSED -> lab.warning
                                            else -> lab.fgSecondary
                                        },
                                    ),
                            )
                            if (index < stops.lastIndex) {
                                Box(
                                    modifier = Modifier
                                        .width(2.dp)
                                        .height(22.dp)
                                        .background(lab.fgTertiary.copy(alpha = 0.35f)),
                                )
                            }
                        }
                        Column(modifier = Modifier.padding(start = 10.dp)) {
                            Text(
                                text = stop.title.ifBlank { stop.id },
                                fontSize = 14.sp,
                                fontWeight = FontWeight.SemiBold,
                                color = lab.fg,
                            )
                            Text(
                                text = stop.state.name.replace('_', ' '),
                                fontSize = 11.sp,
                                fontWeight = FontWeight.Bold,
                                fontFamily = FontFamily.Monospace,
                                color = if (stop.firstClass) lab.destructive else lab.fgSecondary,
                            )
                        }
                    }
                }
            }
        }
    }
}

@Composable
fun FieldMoneyStrip(counts: MoneyHealthCounts, modifier: Modifier = Modifier) {
    val lab = LocalPegasusColors.current
    Row(modifier = modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(8.dp)) {
        listOf(
            "CASH" to counts.pendingCash,
            "FISCAL" to counts.openFiscal,
            "CREDIT" to counts.creditLeave,
        ).forEach { (label, count) ->
            PegasusCard(modifier = Modifier.weight(1f)) {
                Column(Modifier.padding(10.dp)) {
                    Text(label, fontSize = 10.sp, fontWeight = FontWeight.Black, fontFamily = FontFamily.Monospace, color = lab.fgTertiary)
                    Text(
                        if (count == 0) "empty" else count.toString(),
                        fontSize = 13.sp,
                        fontWeight = FontWeight.SemiBold,
                        color = if (count == 0) lab.fgTertiary else lab.fg,
                    )
                }
            }
        }
    }
}

@Composable
fun ManifestBulletMeter(
    state: String?,
    usedVU: Double?,
    maxVU: Double,
    modifier: Modifier = Modifier,
) {
    val lab = LocalPegasusColors.current
    PegasusCard(modifier = modifier.fillMaxWidth()) {
        Column(Modifier.padding(PegasusSpacing.s16), verticalArrangement = Arrangement.spacedBy(6.dp)) {
            Text("MANIFEST", fontSize = 11.sp, fontWeight = FontWeight.Black, fontFamily = FontFamily.Monospace, color = lab.fgTertiary)
            Text(
                if (state.isNullOrBlank()) "unavailable" else state,
                fontSize = 13.sp,
                fontWeight = FontWeight.Bold,
                fontFamily = FontFamily.Monospace,
                color = lab.fg,
            )
            if (usedVU != null && maxVU > 0) {
                LinearProgressIndicator(
                    progress = (usedVU / maxVU).toFloat().coerceIn(0f, 1f),
                    modifier = Modifier.fillMaxWidth(),
                )
                Text("%.0f / %.0f VU".format(usedVU, maxVU), fontSize = 11.sp, fontFamily = FontFamily.Monospace, color = lab.fgSecondary)
            } else {
                Text("VU unavailable", fontSize = 11.sp, color = lab.fgTertiary)
            }
        }
    }
}
